analyzer_bin_envvar used to get analyzer binary in runtime

sibeream opened the pull request:
Modules can define analyzer_bin_envvar field as non-NULL. If it's done, an environment variable with the same name as the value of the field will be used to determine the path to the analyzer binary.

sibeream committed:
analyzer_bin_envvar used to get analyzer binary in runtime

kdudka commented on src/csgcca.c:
```diff
@@ -37,6 +37,8 @@ const char *wrapper_debug_envvar_name = "DEBUG_CSGCCA";
 
 const char *analyzer_name = "gcc";
 
+const char *analyzer_bin_envvar = "CSGCCA_ANALYZER_BIN";
```
Could you please document the variable here?

https://github.com/csutils/cscppc/blob/f02ccc233408a3b8cbc9b31989a459acfacdf567/doc/csgcca.adoc#environment-variables

kdudka commented on src/cswrap-core.c:
```diff
@@ -393,8 +393,14 @@ static void consider_running_analyzer(
+    if (!analyzer || !analyzer[0])
+        analyzer = analyzer_name;
+
     /* make sure that the analyzer process is named analyzer_name */
```
Could you please update the comment to reflect the code change?

kdudka commented on src/cswrap-core.h:
```diff
@@ -34,6 +34,21 @@ extern const char *wrapper_debug_envvar_name;
+ * run analyzer, just the binary it self.
+ */
+extern const char *analyzer_bin_envvar;
+
```
Thank you for documenting the new symbol! I propose to simplify the text and clarify that the shell variable takes a relative or absolute path to the analyzer's executable.  And please rewrap the text at column 80.

kdudka reviewed (changes requested):
Thanks!  It works for me.  I suggest a few minor changes inline.

kdudka commented on src/csclng.c:
```diff
@@ -37,6 +37,8 @@ const char *wrapper_debug_envvar_name = "DEBUG_CSCLNG";
 
 const char *analyzer_name = "clang";
 
+const char *analyzer_bin_envvar = NULL;
```
Could you please name the new symbol `analyzer_bin_envvar_name`?  It would be more consistent with the existing symbols `wrapper_addopts_envvar_name` and `wrapper_debug_envvar_name`.

sibeream committed:
Changes after review.

1. analyzer_bin_envvar renamed  to analyzer_bin_envvar_name
2. CSGCCA_ANALYZER_BIN documented
3. comments in code are updated

kdudka commented:
Thanks for the quick turnaround!  Merging...
